src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -253,7 +277,7 @@ void containers(Context context, DeclarationList declList) throws ParserExceptio
                     dcl.definition.text = "\n" + dcl.definition.text;
                     decl.declarator = dcl;
                 }
-                LinkedHashSet<Type> typeSet = new LinkedHashSet<>();
+                LinkedList<Type> typeSet = new LinkedList<>();
                 typeSet.addAll(Arrays.asList(firstType, secondType, indexType, valueType));
                 typeSet.addAll(Arrays.asList(containerType.arguments));
```

Comment:
Please revert that, as well as the unnecessary import. That's intentional to prevent duplicates. Please test your changes against all the presets!

Reply:
This is related to the addition of `hashCode` in `Type`. Before, there was no `hashCode`, so the `LinkedHashSet` was functionally more similar to a `LinkedList`. I added a `hashCode`, consistent with existing `equals`, because I needed a `Set<Type>` elsewhere. So this change in fact keeps the previous behavior.
I did test against all the presets, and in effect without this change I had parsing errors or changes. I didn't investigate further why `typeSet` must not de-duplicate types. We can do it if you want.

Reply:
Without the hashCode, the effect of using a Set was to remove "exact" duplicates. That is, if `t1 == t2`, it keeps only one. With the hashCode, it removes duplicates that are the same in terms of `equal`, that is that have the same cppName and javaName.
When, for instance, `firstType` and `secondType` are 2 distinct instances of `Type` but with the same cppName, we need to process each of them in the following loop to add the appropriate `annotation`. That's why it didn't work anymore after the addition of `hashCode`.
After the change from a `Set` to a `List`, if `firstType` and `secondType` are the same instance, the instance is processed twice, but that's not a big deal since, if I'm not wrong, there are tests in the body of the loop that prevents the annotations to be changed twice.

Reply:
There's no reason to use LinkedList though, so let's use ArrayList, and let's do the same for the other LinkedHashSet

Reply:
> There's no reason to use LinkedList though, so let's use ArrayList,

Or a plain array...
Done.

> and let's do the same for the other LinkedHashSet

`infoSet` ?
I'm not sure. are you ?
The `addAll` could result in the addition of the same `Info` several times, and executing the iterations of the loop several times for the same Info seems to have real effects.